src/main/context/AuthContext.jsx

```diff
@@ -0,0 +1,42 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+import { GoogleAuthProvider, signInWithPopup, signOut, onAuthStateChanged } from "firebase/auth";
+import { auth } from "../firebase";
+
+const AuthContext = createContext()
+
+export const AuthContextProvider = ({ children }) => {
+  const [user, setUser] = useState()
+
+  // google signin
+  const googleSignIn = () => {
+    const provider = new GoogleAuthProvider();
+    signInWithPopup(auth, provider)
+  }
+
+  // signout
+  const logOut = () => {
+    signOut(auth)
+  }
+
+  // onAuthChange
+  useEffect(() => {
+    const unSubscribe = onAuthStateChanged(auth, (currentUser) => {
+      setUser(currentUser)
```

Comment:
after getting currentUser, collect necessary data from currentUser and only set those details in the user state and store the full currentUser Details in a seperate useState like firebaseUser

_necessary data eg: firstName, lastName, email, imageUrl ...etc_

Reply:
created **firebaseUser** and added full **currentUser** details to it

changed user data to new obj 

![image](https://github.com/Anuraj-TR/spendlux/assets/101694416/39c2d925-4f0a-4b77-98a0-732912359f0e)